tests/Microsoft.Identity.Web.Test/TokenAcquisitionAuthorityTests.cs

```diff
@@ -309,5 +309,109 @@ public void MergeExtraQueryParameters_MergedOptionsNull_Test()
             // Assert
             Assert.Null(mergedDict);
         }
+
+        [Theory]
+        [InlineData("https://localhost:1234")]
+        [InlineData("")]
+        [InlineData(null)]
+        public void ManagedIdCacheKey_Test(string? clientId)
+        {
+            // Arrange
+            string defaultKey = "SYSTEM";
+            ManagedIdentityOptions managedIdentityOptions = new()
+            {
+                UserAssignedClientId = clientId
+            };
+
+            // Act
+            string key = TokenAcquisition.GetCacheKeyForManagedId(managedIdentityOptions);
+
+            // Assert
+            if (string.IsNullOrEmpty(clientId))
+            {
+                Assert.Equal(defaultKey, key);
+            }
+            else
+            {
+                Assert.Equal(clientId, key);
+            }
+        }
+
+        [Theory]
+        [InlineData("https://localhost:1234")]
+        [InlineData("")]
+        [InlineData(null)]
+        public async void GetOrBuildManagedIdentity_TestAsync(string? clientId)
+        {
+            // Arrange
+            ManagedIdentityOptions managedIdentityOptions = new()
+            {
+                UserAssignedClientId = clientId
+            };
+            MergedOptions mergedOptions = new();
+            BuildTheRequiredServices();
+            InitializeTokenAcquisitionObjects();
+
+            // Act
+            IManagedIdentityApplication app1 = 
```

Comment:
>IManagedIdentityApplication [](http://example.com/codeflow?start=12&length=27)

as mentioned earlier, this creates an additional allocation for interface, so just use var...I know it's less reader friendly, but just a good habit for perf. PS> i know this is a test :)

Reply:
@jennyf19, right now my IDE is upset when I use var in this situation, do we have recommended settings so the IDE lines up with our preferred usage? Here are [the preference settings](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0007-ide0008).